apps/tidl_deploy/tidl_import.py

```diff
@@ -0,0 +1,272 @@
+* DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+* OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+* OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+* OF THE POSSIBILITY OF SUCH DAMAGE.
+"""
+
+import os
+import subprocess
+import numpy as np
+import cv2
+import tensorflow as tf
+from tensorflow.python.framework import dtypes
+from tensorflow.python.tools import optimize_for_inference_lib
+
+################################################################################
+# Function tidl_check_model: 
+#   Check if a given model is supported by TIDL. Currently, only Tensorflow
+#   models can be recognized by this script. 
+# Input:
+#       model_file:        file name of the given model
+#       calib_image:       file name of image for TIDL calibration
+#       subgraph_name:     base name of TIDL network and params binary files 
+#       model_input_shape: shape of the input tensor of the model 
+#       tidl_import_tool:  file name of TIDL import tool executable
+#       tidl_calib_tool:   file name of TIDL calibration tool executable
+# Output: 
+#       "subgraph_name"_net.bin:    TIDL network binary file
+#       "subgraph_name"_params.bin: TIDL parameters binary file
+# Return:
+#       1 if the model can run on TIDL 
+#       0 if the model cannot run on TIDL
+#
+# Note: Input tensor shape is not available from every supported model. For 
+#       example, mobileNet v2 frozen model doesn't have the shape information. 
+#       Therefore, it needs to be provided by the caller. 
+################################################################################
+def tidl_check_model(model_file, calib_image, subgraph_name, model_input_shape,
+                     tidl_import_tool, tidl_calib_tool, artifacts_folder, conv2d_kernel_type):
+
+  print('Check if TIDL supports ' + model_file)
+
+  # Check model type - can only recognize TF models
+  if model_file.endswith('.pb'):
+    print ('Importing TF model to TIDL')
+  else:
+    print ('This model is not supported by TIDL')
+    return 0
+
+  # Optimize TF model - assuming provided model is a frozen model
+  opt_model_file = model_file.replace(".pb", "_opt.pb")
+  result = tidl_optimize_tf_model(model_file, opt_model_file)
+  if(result == 0):
+    return 0
+
+  # Pre-process calibration image to raw data saved in binary format
+  raw_image = "./raw_calib_image.bin"
+  tidl_tf_image_preprocess(calib_image, raw_image, model_input_shape)
+
+  # Try to import the model: return 1 if import succeeds and 0 if import fails
+  result = tidl_import_tf_model(opt_model_file, raw_image, model_input_shape,
+                                       tidl_import_tool, tidl_calib_tool, subgraph_name, artifacts_folder, conv2d_kernel_type)
+
+  return result
+
+################################################################################

```

Comment:
please follow the tvm docstring here: https://docs.tvm.ai/contribute/code_guide.html
This will help sphinx to parse it.

One of the examples can be found here:
https://github.com/apache/incubator-tvm/blob/master/python/tvm/relay/build_module.py#L178